backend/views.py

```diff
@@ -120,3 +123,67 @@ class SongsDatatableView(BaseDatatableView):
     model = Song
     max_display_length = 500
     columns = ["name", "author", "capo"]
+
+
+# pylint: disable=unused-argument
+def quelea_song_import(uploaded_file, cleaned_data):
+    """Read data from file in cleaned_data, and import all songs in qsp archive"""
+
+    def read_quelea_file(filename):
+        with zipfile.ZipFile(filename) as file:
+            return {name: file.read(name) for name in file.namelist()}
+
+    def get_lyrics(mapp):
+        ret = []
+        for name in mapp.keys():
+            song = ET.fromstring(mapp[name])
+            lyrics_lines = []
+            for child in song.findall('.//lyrics'):
+                if child.text is not None:
+                    lyrics_lines.append(child.text)
+
+            ret.append((name, '\n\n'.join(lyrics_lines)))
+        return ret
+
+    songs = read_quelea_file(cleaned_data['file'])
+    lyric = get_lyrics(songs)
+
+    songs = 0
+    for (name, text) in lyric:
+        clean_name = name.rstrip('.xml')
+
+        song_obj = Song(name=clean_name, text=text)
+        song_obj.save()
+
+        list_ids = [f.pk for f in cleaned_data['categories']]
+
+        song_obj.categories.set(list_ids)
+        songs += 1
+
+    return songs
+
+
+@method_decorator(login_required, name='dispatch')
+class UploadView(FormView):
+    """View to recieve qsp file and import all songs"""
+
+    form_class = UploadFileForm
+    template_name = 'songs/import.html'
+
+    success_url = reverse_lazy('backend:index')
+
+    # def get_success_url(self):
+    #     regenerate_pdf(self.object)
+    #     regenerate_prerender(self.object)
+    #     return super().get_success_url()
+
+    def post(self, request, *args, **kwargs):
```

Comment:
You could try using `form_valid()` method instead https://docs.djangoproject.com/en/4.0/topics/class-based-views/generic-editing/#basic-forms

Reply:
implemented.